---
title: "Deng single cell analysis"
author: "Kushal K Dey"
date: "October 8, 2015"
output: 
  html_document:
    css: floating-toc.css
    toc: true
---

## Objective

In this script, we perform the Grade of Membership model (GoM) on the dataset obtained from the study due to Deng \emph{et al}. Check the [paper](http://science.sciencemag.org/content/343/6167/193).

The dataset has single cells sequenced starting from zygote through the 2 cell, 4 cell all the way up to blastocyst. We want to cluster the cells by  development phase, also taking into account the information about the embryo of the origin of the cells and try to see which genes are actually driving the clusters in the development phase. These would be the genes that would also play a significant part in differentiating the various stages of the development.


```{r echo=TRUE, eval=TRUE, warning=FALSE, message=FALSE}
rm(list=ls())
library(data.table)
#install_github('kkdey/maptpx') 
library(maptpx)
library(CountClust)
library(data.table)

```


## Prepare the Data

```{r echo=TRUE, eval=FALSE}
files <- list.files("../external_data/Deng_Data/Deng_files/");

num <- 1
temp_data <- data.frame(fread(paste0('../external_data/Deng_Data/Deng_files/',files[num])));
gene_names <- temp_data$X.Gene_symbol;

reads_mat <- cbind.data.frame(gene_names);

for(num in 1:length(files))
{
  temp_data <- data.frame(fread(paste0('../data/Deng_data/Deng_files/',files[num])));
  reads_mat <- cbind.data.frame(reads_mat,temp_data$reads);
}

cell_meta <- unlist(lapply(files, function(x) strsplit(x,"_")[[1]][2]));
colnames(reads_mat) <- c("gene_names",files);
reads_no_dups <- reads_mat %>%
                  group_by(gene_names) %>%
                  summarise_each(funs(sum))

reads_no_dups <- data.frame(reads_no_dups)

gene_names_new <- reads_no_dups[,1]
reads_no_dups <- reads_no_dups[,-1];
rownames(reads_no_dups) <- gene_names_new;
colnames(reads_no_dups) <- cell_meta;
dim(reads_no_dups);

write.table(reads_no_dups,"../external_data/Deng_cell_data.txt");
```

Now we load the data.

## Data and Meta-data Processing

```{r echo=TRUE, eval=FALSE}
reads <- data.frame(fread('../external_data/Deng_Data/Deng_cell_data.txt'),row.names=1);
files <- list.files("../external_data/Deng_Data/Deng_files/");
cell_meta <- unlist(lapply(files, function(x) strsplit(x,"_")[[1]][2]));
embryo_id <- unlist(lapply(files, function(x) strsplit(strsplit(x,"_")[[1]][3],"-")[[1]][1]));
cell_meta[grep("zy",cell_meta)]="zy";
cell_meta[grep("smartseq2", files)]="8cell_nd";
cell_meta[grep("8cell_2pooled", files)]="8cell_nd";
cell_meta[grep("8cell_split", files)]="8cell_nd";
cell_meta[grep("16cell_2pooled", files)]="16cell_nd";
cell_meta[grep("16cell_split", files)]="16cell_nd";
indices_not_reqd <- which(cell_meta=="BXC"   | cell_meta=="C57twocell" | cell_meta=="fibroblast" | cell_meta =="8cell_nd" | cell_meta == "16cell_nd");
cell_meta <- cell_meta[-indices_not_reqd];
embryo_id <- embryo_id[-indices_not_reqd];
embryo_id[which(embryo_id == "expression.txt")]="."
cell_embryo <- paste0(cell_meta,"_", embryo_id);
reads <- reads[,-indices_not_reqd];
cell_meta_unique <- c("zy","early2cell","mid2cell","late2cell","4cell","8cell","16cell","earlyblast","midblast","lateblast") ;
order_of_development <- order(match(cell_meta,cell_meta_unique))
reads <- reads[,order_of_development];
cell_meta <- cell_meta[order_of_development]
cell_embryo <- cell_embryo[order_of_development];
colnames(reads) <- cell_meta;
```

```{r echo=TRUE, eval=TRUE}
load("../rdas/deng_topic_fit.rda")

# extract the omega matrix: membership weights of each cell
names(Topic_clus_list)
str(Topic_clus_list$clust_6)
omega <- Topic_clus_list$clust_6$omega

# import embryonl labels
embryo_label <- read.table("../external_data/Deng_Data/cell_labels_phase_embryo.txt",
                           quote = "\"",
                           header = TRUE,
                           stringsAsFactors = FALSE)$x
head(embryo_label, 20)
table(embryo_label)
stopifnot(length(embryo_label) == NROW(omega))


# make annotation matrix for the plot of all tissues
# sample_id has to be unique
annotation <- data.frame(
  sample_id = paste0("X", c(1:NROW(omega))),
  tissue_label = factor(rownames(omega),
                        levels = rev( c("zy", "early2cell", "mid2cell", "late2cell",
                                       "4cell", "8cell", "16cell", "earlyblast",
                                       "midblast", "lateblast") ) ) )

# make annotation for early stage plot
# sample_id has to be unique
annotation_embryo <- data.frame(
    sample_id = paste0("X", c(1:NROW(omega))),
    tissue_label = factor(embryo_label,
      levels = rev( c("zy_.",
                      paste("early2cell",c("0r", c(1:3)), sep = "_"),
                      paste("mid2cell",c("0r", c(3:7)), sep = "_"),
                      paste("late2cell",c("0r", c(5:9)), sep = "_"),
                      paste("4cell",c("0r", c(1:4)), sep = "_"),
                      paste("8cell",c("0r", c(1,2,5,8)), sep = "_"),
                      paste("16cell",c("0r", c(1,4,5,6)), sep = "_"),
                      paste("earlyblast",c("0r", c(2:4)), sep = "_"),
                      paste("midblast",c("0r", c(1:3)), sep = "_"),
                      paste("lateblast",c("0r", c(1:3)), sep = "_") ) ) ) )




# after extracting tissue type of each sample
# recode each sample to have unique rownames
rownames(omega) <- paste0("X", annotation$sample_id)
```

## Structure plot of Deng data(K=6)

```{r echo=TRUE, eval=TRUE, fig.height=7, fig.width=5, message=FALSE, warning=FALSE}

StructureGGplot(omega = omega,
               annotation = annotation,
               palette = RColorBrewer::brewer.pal(8, "Accent"),
               figure_title = "",
               yaxis_label = "Cell type",
               sample_order_decreasing = FALSE,
               axis_tick = list(axis_ticks_length = .1,
                                 axis_ticks_lwd_y = .1,
                                 axis_ticks_lwd_x = .1,
                                 axis_label_size = 7,
                                 axis_label_face = "bold"))

```

