---
title: "Admixture analysis tests"
author: "Kushal K Dey"
date: "July 3, 2015"
output: html_document
---

```{r packages}
library(maptpx)
library(qtlcharts)
library(data.table)

```

## GTEx Thinned Data

We thinned the gtex data so that the number of reads per tissue sample and per gene is few and of the order of the single cell level data. In the Jaitin et al dataset, there were on average, 2540 reads on an average and in the full experiment with 4590 cells and 20190 genes, the toal number of reads recorded was 11658921. On the other hand, for the GTEx dataset, the corresponding number was way bigger. For the 16407 cis genes, there were 3.5e+11 reads and for each sample on an average, there were 41073880 reads, which is around $10^4$ larger than the single cell experiment reads in the Jaitin dataset. So, we use two thinning thresholds, $p=0.0001$ and $p=0.00001$. 

### Gtex- thinned Structure (Thinning threshold 0.001)

```{r echo=FALSE, eval=TRUE}
docweights_thinned_version_1 <- as.matrix(read.table("/Users/kushal/Documents/gtex-viz/gtex.Kushal/data/topics_omega_clus_12_version_1.txt"));
K=12;
color=c("red","blue","cornflowerblue","black","cyan","darkblue",
        "brown4","burlywood","darkgoldenrod1","darkgray","deepskyblue","darkkhaki",
        "firebrick","darkorchid","hotpink","green","magenta","yellow", "azure1","azure4");


samples_id=read.table("/Users/kushal/Documents/gtex-viz/gtex.Kushal/data/samples_id.txt")[,3];

docweights_per_tissue_mean <- apply(docweights_thinned_version_1,2,function(x) tapply(x,samples_id,mean));
ordering=heatmap(docweights_per_tissue_mean, col=gray.colors(200))$rowInd;

unique_samples_id_ordered = unique(samples_id)[ordering];


clus_ordered =unlist(lapply(1:53, function(x) which(samples_id == unique_samples_id_ordered[x])));


docweights_ordering = docweights_thinned_version_1[clus_ordered,];
samples_id_ordered = samples_id[clus_ordered];
par(mar=c(14,3,3,3))
barplot(t(docweights_ordering),col=color[1:K],axisnames=F,space=0,border=NA,main=paste("No. of clusters=",K),las=1,ylim=c(0,1),cex.axis=1.5,cex.main=1.4)

labels = match(unique(samples_id_ordered), samples_id_ordered);
abline(v=labels)

labels_low=labels;
labels_up=c(labels[2:length(labels)],dim(docweights_ordering)[1]);
mid_point=labels_low +0.5*(labels_up-labels_low);

axis(1,at=mid_point, unique(samples_id_ordered),las=2);


```

### Gtex- thinned Structure (Thinning threshold 0.0001)

```{r echo=FALSE, eval=TRUE}

docweights_thinned_version_2 <- as.matrix(read.table("/Users/kushal/Documents/gtex-viz/gtex.Kushal/data/topics_omega_clus_12_version_2.txt"));
K=12;
color=c("red","blue","cornflowerblue","black","cyan","darkblue",
        "brown4","burlywood","darkgoldenrod1","darkgray","deepskyblue","darkkhaki",
        "firebrick","darkorchid","hotpink","green","magenta","yellow", "azure1","azure4");


samples_id=read.table("/Users/kushal/Documents/gtex-viz/gtex.Kushal/data/samples_id.txt")[,3];

docweights_per_tissue_mean <- apply(docweights_thinned_version_2,2,function(x) tapply(x,samples_id,mean));
ordering=heatmap(docweights_per_tissue_mean, col=gray.colors(200))$rowInd;

unique_samples_id_ordered = unique(samples_id)[ordering];

clus_ordered =unlist(lapply(1:53, function(x) which(samples_id == unique_samples_id_ordered[x])));


docweights_ordering = docweights_thinned_version_2[clus_ordered,];
samples_id_ordered = samples_id[clus_ordered];
par(mar=c(14,3,3,3))
barplot(t(docweights_ordering),col=color[1:K],axisnames=F,space=0,border=NA,main=paste("No. of clusters=",K),las=1,ylim=c(0,1),cex.axis=1.5,cex.main=1.4)

labels = match(unique(samples_id_ordered), samples_id_ordered);
abline(v=labels)

labels_low=labels;
labels_up=c(labels[2:length(labels)],dim(docweights_ordering)[1]);
mid_point=labels_low +0.5*(labels_up-labels_low);

axis(1,at=mid_point, unique(samples_id_ordered),las=2);

```

### How the Brain looks? (thinned at 0.001)

```{r echo=FALSE, eval=TRUE}

samples_id=read.table("/Users/kushal/Documents/gtex-viz/gtex.Kushal/data/samples_id.txt")
brain_indices=which(samples_id[,2]=='Brain');
brain_subgroups_id =samples_id[brain_indices,3];

temp <- sapply(1:length(brain_subgroups_id), function(n)  strsplit(as.character(brain_subgroups_id[n]),' - ')[[1]][2])

brain_subgroups_id <- temp;


docweights= read.table('/Users/kushal/Documents/gtex-viz/gtex.Kushal/data/topics_brain_thinned_omega_clus_4.txt');

K=4
png(filename = '../plots/brain_structure_plot.png')
par(mar=c(12,4,4,4))
barplot(t(docweights),col=2:(K+1),axisnames=F,space=0,border=NA,main=paste("Structure plot for brain samples:K=",K),las=1,ylim=c(0,1),cex.axis=1.5,cex.main=1.4)
brain_subgroups_id[394]=brain_subgroups_id[395];
labels=match(unique(brain_subgroups_id), brain_subgroups_id);
abline(v=labels)

labels_low=labels;
labels_up=c(labels[2:length(labels)],dim(docweights)[1]);
mid_point=labels_low +0.5*(labels_up-labels_low);

axis(1,at=mid_point, unique(brain_subgroups_id),las=2);
dev.off()




```

### Correlation heatmap of brain after thinning

```{r echo=FALSE, eval=TRUE}

docweights= read.table('/Users/kushal/Documents/gtex-viz/gtex.Kushal/data/topics_brain_thinned_omega_clus_4.txt');

brain_topics_prop_per_tissue = apply(docweights, 2, function(x) tapply(x,as.factor(as.vector(as.matrix(brain_subgroups_id))),mean));

iplotCorr(t(brain_topics_prop_per_tissue),chartOpts=list(corcolors = c("white", "white","white", "crimson"),zlim=c(0,0.25,0.6,1),margin = list(left=150, top=40, right=3, bottom=70, inner=5)),reorder=TRUE)

```

### Correlation heatmap of brain without admixture

```{r echo=FALSE, eval=TRUE}
data= data.frame(fread('/Users/kushal/Documents/gtex-viz/gtex.Kushal/data/gtex_thinned_version_1.txt'))[,-1];
brain_indices=which(samples_id[,2]=='Brain');
brain_samples=data[,brain_indices];

brain_subgroups_id =as.vector(brain_subgroups_id);
brain_samples_mean_per_tissue = apply(brain_samples, 1, function(x) tapply(x,as.factor(brain_subgroups_id),mean));

iplotCorr(t(brain_samples_mean_per_tissue),chartOpts=list(corcolors = c("white", "white","white", "crimson"),zlim=c(0,0.25,0.6,1),margin = list(left=150, top=40, right=3, bottom=70, inner=5)),reorder=TRUE)

```

### What about the full thinned data??....admixture first

Thinning parameter again $0.001$.

```{r echo=FALSE, eval=TRUE}

docweights_thinned_version_1 <- as.matrix(read.table("/Users/kushal/Documents/gtex-viz/gtex.Kushal/data/topics_omega_clus_12_version_1.txt"));

subgroups_id=read.table("/Users/kushal/Documents/gtex-viz/gtex.Kushal/data/samples_id.txt")[,3];

topics_prop_per_tissue = apply(docweights_thinned_version_1, 2, function(x) tapply(x,as.factor(as.vector(as.matrix(subgroups_id))),mean));

iplotCorr(t(topics_prop_per_tissue),chartOpts=list(corcolors = c("white", "white","white", "crimson"),zlim=c(0,0.25,0.6,1),margin = list(left=150, top=40, right=3, bottom=70, inner=5)),reorder=TRUE)


```

### Full thinned data hierarchical clustering

```{r echo=FALSE, eval=TRUE}

data= data.frame(fread('/Users/kushal/Documents/gtex-viz/gtex.Kushal/data/gtex_thinned_version_1.txt'))[,-1];

subgroups_id=read.table("/Users/kushal/Documents/gtex-viz/gtex.Kushal/data/samples_id.txt")[,3];

samples_mean_per_tissue = apply(data, 1, function(x) tapply(x,as.factor(subgroups_id),mean));

iplotCorr(t(samples_mean_per_tissue),chartOpts=list(corcolors = c("white", "white","white", "crimson"),zlim=c(0,0.25,0.6,1),margin = list(left=150, top=40, right=3, bottom=70, inner=5)),reorder=TRUE)


```

Finally..Are there actually 2 clusters in Liver? (remember t-SNE!!)

```{r echo=FALSE, eval=TRUE}

data= data.frame(fread("/Users/kushal/Documents/gtex-viz/gtex.Kushal/data/cis_gene_expression_all_samples.txt"))[,-1];
samples_id=read.table("/Users/kushal/Documents/gtex-viz/gtex.Kushal/data/samples_id.txt");
liver_indices=which(samples_id[,3]=='Liver');

liver_samples=data[,liver_indices];

liver_samples_temp= as.matrix(liver_samples);

liver_samples_thinned =apply(liver_samples_temp,c(1,2), function(x) rbinom(1,x,prob=0.0001));
Topic_Clus=topics(t(liver_samples_thinned),K=2, tol=0.001);
K=2
docweights=Topic_Clus$omega;
par(mar=c(10,4,4,2))
barplot(t(docweights),col=2:(K+1),axisnames=F,space=0,border=NA,main=paste("No. of clusters=",K),las=1,ylim=c(0,1),cex.axis=1.5,cex.main=1.4)



```

Does not look like any distinct clusters are present.

## Single cell data analysis 

### Eberwine Data

[Paper](http://www.ncbi.nlm.nih.gov/pubmed/26056000)

mouse single cells

```{r eberwine, echo=FALSE, eval=TRUE}
docweights = as.matrix(read.table("/Users/kushal/Documents/Matthew Stephens Project/counts_clustering/Topic_omega_clus_10.txt"));
K=10;
barplot(t(docweights),col=2:(K+1),axisnames=F,space=0,border=NA,main=paste("No. of clusters=",K),las=1,ylim=c(0,1),cex.axis=1.5,cex.main=1.4)
listFiles <- Sys.glob("/Users/kushal/Documents/Matthew Stephens Project/counts_clustering/GSE56638_RAW/GSM1366*.txt");
cell_names_temp = unlist(lapply(listFiles, function(x) substring(x,92,95)));
cell_names = unlist(lapply(strsplit(cell_names_temp, split='_', fixed=TRUE), function(x) x[1]));
labels=match(unique(cell_names), cell_names);
abline(v=labels-1)
labels_low=labels;
labels_up=c(labels[2:length(labels)],dim(docweights)[1]);
mid_point=labels_low +0.5*(labels_up-labels_low);
axis(1,at=mid_point, unique(cell_names),las=2);

```


### Satija Data 

[Paper](http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE66688)

Zebrafish single cell data

```{r, echo=FALSE, eval=TRUE}
K=10;
docweights=as.matrix(read.table(paste0("/Users/kushal/Documents/Matthew Stephens Project/counts_clustering/data/topics_omega_Satija_clus",K)));


cell_names = as.matrix(read.table("/Users/kushal/Documents/Matthew Stephens Project/counts_clustering/data/cell_names_GSE66688"))

cell_names_temp = unlist(lapply(strsplit(as.vector(cell_names),"_"),function(x) paste0(x[1],"_",x[2])));

cell_names_temp_ordered = cell_names_temp[order(cell_names_temp)];

docweights_ordered= docweights[order(cell_names_temp),];

barplot(t(docweights_ordered),col=2:(K+1),axisnames=F,space=0,border=NA,main=paste("No. of clusters=",K),las=1,ylim=c(0,1),cex.axis=1.5,cex.main=1.4)

labels=match(unique(cell_names_temp_ordered), cell_names_temp_ordered);
abline(v=labels-1)
labels_low=labels;
labels_up=c(labels[2:length(labels)],dim(docweights_ordered)[1]);
mid_point=labels_low +0.5*(labels_up-labels_low);
axis(1,at=mid_point, unique(cell_names_temp_ordered),las=2);

```

### Mouse cortex + hippocampus 

[Paper](http://www.sciencemag.org/content/347/6226/1138.short)

```{r, echo=FALSE, eval=TRUE}
data=suppressWarnings(suppressMessages(data.frame(fread("/Users/kushal/Documents/Matthew Stephens Project/counts_clustering/expression_mRNA_17-Aug-2014.txt"))));

counts_data=as.matrix(data[-(1:10),-(1:2)]);

K=2;
docweights=as.matrix(read.table(paste0("/Users/kushal/Documents/Matthew Stephens Project/counts_clustering/data/topics_omega_Amit_clus",K)));

cell_type = unlist(lapply(strsplit(colnames(counts_data),"\\."),function(x) x[1]));

cell_type_ordered = cell_type[order(cell_type)];

docweights_ordered= docweights[order(cell_type),];

barplot(t(docweights_ordered),col=2:(K+1),axisnames=F,space=0,border=NA,main=paste("No. of clusters=",K),las=1,ylim=c(0,1),cex.axis=1.5,cex.main=1.4)

labels=match(unique(cell_type_ordered), cell_type_ordered);
abline(v=labels-1)
labels_low=labels;
labels_up=c(labels[2:length(labels)],dim(docweights_ordered)[1]);
mid_point=labels_low +0.5*(labels_up-labels_low);
axis(1,at=mid_point, unique(cell_type_ordered),las=2);

subtype = data[8,-(1:2)];
subtype_ordered = subtype[order(subtype)];

docweights_subtype_ordered = docweights[order(subtype),];

barplot(t(docweights_subtype_ordered),col=2:(K+1),axisnames=F,space=0,border=NA,main=paste("No. of clusters=",K),las=1,ylim=c(0,1),cex.axis=1.5,cex.main=1.4)

labels=match(unique(as.vector(as.matrix(subtype_ordered))), subtype_ordered);
abline(v=labels-1)
labels_low=labels;
labels_up=c(labels[2:length(labels)],dim(docweights_ordered)[1]);
mid_point=labels_low +0.5*(labels_up-labels_low);
axis(1,at=mid_point, unique(as.vector(as.matrix(subtype_ordered))),las=2);

```

The Structure plot for 10 clusters for the dataset is

```{r echo=FALSE, eval=TRUE}

K=10;
docweights=as.matrix(read.table(paste0("/Users/kushal/Documents/Matthew Stephens Project/counts_clustering/data/topics_omega_Amit_clus",K)));

par(mar=c(10,4,4,1))
barplot(t(docweights),col=2:(K+1),axisnames=F,space=0,border=NA,main=paste("No. of clusters=",K),las=1,ylim=c(0,1),cex.axis=1.5,cex.main=1.4)

cell_type = unlist(lapply(strsplit(colnames(counts_data),"\\."),function(x) x[1]));

cell_type_ordered = cell_type[order(cell_type)];

docweights_ordered= docweights[order(cell_type),];

barplot(t(docweights_ordered),col=2:(K+1),axisnames=F,space=0,border=NA,main=paste("No. of clusters=",K),las=1,ylim=c(0,1),cex.axis=1.5,cex.main=1.4)

labels=match(unique(cell_type_ordered), cell_type_ordered);
abline(v=labels-1)
labels_low=labels;
labels_up=c(labels[2:length(labels)],dim(docweights_ordered)[1]);
mid_point=labels_low +0.5*(labels_up-labels_low);
axis(1,at=mid_point, unique(cell_type_ordered),las=2);

subtype = data[8,-(1:2)];
subtype_ordered = subtype[order(subtype)];

docweights_subtype_ordered = docweights[order(subtype),];

barplot(t(docweights_subtype_ordered),col=2:(K+1),axisnames=F,space=0,border=NA,main=paste("No. of clusters=",K),las=1,ylim=c(0,1),cex.axis=1.5,cex.main=1.4)

labels=match(unique(as.vector(as.matrix(subtype_ordered))), subtype_ordered);
abline(v=labels-1)
labels_low=labels;
labels_up=c(labels[2:length(labels)],dim(docweights_ordered)[1]);
mid_point=labels_low +0.5*(labels_up-labels_low);
axis(1,at=mid_point, unique(as.vector(as.matrix(subtype_ordered))),las=2);

pooled_type = paste0(cell_type,"_",subtype);
pooled_type_ordered = pooled_type[order(pooled_type)];

docweights_pooled_type_ordered = docweights[order(pooled_type),];

barplot(t(docweights_pooled_type_ordered),col=2:(K+1),axisnames=F,space=0,border=NA,main=paste("No. of clusters=",K),las=1,ylim=c(0,1),cex.axis=1.5,cex.main=1.4)

labels=match(unique(as.vector(as.matrix(pooled_type_ordered))), pooled_type_ordered);
abline(v=labels-1)
labels_low=labels;
labels_up=c(labels[2:length(labels)],dim(docweights_pooled_type_ordered)[1]);
mid_point=labels_low +0.5*(labels_up-labels_low);
axis(1,at=mid_point, unique(as.vector(as.matrix(pooled_type_ordered))),las=2);

```

### Finer cell sub types in the Amit data 

```{r echo=FALSE, eval=TRUE}

K=10;
docweights=as.matrix(read.table(paste0("/Users/kushal/Documents/Matthew Stephens Project/counts_clustering/data/topics_omega_Amit_clus",K)));

finer_subtype = data[9,-(1:2)];
finer_subtype_ordered = finer_subtype[order(finer_subtype)];

none_indices = which(finer_subtype_ordered=="(none)");

docweights_ordered = docweights[order(finer_subtype),];

docweights_non_null_ordered = docweights_ordered [-none_indices,];

finer_subtype_non_null_ordered = finer_subtype_ordered[-none_indices];

par(mfrow=c(1,1))
barplot(t(docweights_non_null_ordered),col=2:(K+1),axisnames=F,space=0,border=NA,main=paste("No. of clusters=",K),las=1,ylim=c(0,1),cex.axis=1.5,cex.main=1.4)

labels=match(unique(as.vector(as.matrix(finer_subtype_non_null_ordered))), finer_subtype_non_null_ordered);
abline(v=labels-1)
labels_low=labels;
labels_up=c(labels[2:length(labels)],dim(docweights_non_null_ordered)[1]);
mid_point=labels_low +0.5*(labels_up-labels_low);
axis(1,at=mid_point, unique(as.vector(as.matrix(finer_subtype_non_null_ordered))),las=2);




```

