\section{Methods and Materials}

\subsection{Data preprocessing}

We assume that the data from an RNA-seq experiment have been summarized by a table of counts $C_{N \times G} = ((c_{ng}))$, where $c_{ng}$ is the number of RNA-seq reads from sample $n$ that mapped to gene (or transcript) $g$. Such a table of counts is obtained by processing the BAM or FASTQ files obtained from sequencing machines and currently, there is an effort to make such reads table, ready for statistical analysis, publicly available (check the ReCounts website\cite{Frazee2011}). Before applying the graded membership model on the reads data, we remove the genes with 0 or same count of matched reads across all samples as they are non-informative for clustering. We also remove any sample or gene with NA values of reads and  spike-in control genes,  as the latter may create bias due to their typically having high number of reads mapped to them \cite{Jiang2011}. 

%RNA-seq experiments provide us with a set of FASTQ files that contain the nucleotide sequence of each read and a quality score at each position, which can be mapped to  reference genome or exome or transcriptome. The output of this mapping is usually saved in a SAM/BAM file using SAMtools  \cite{Li2009}, a task primarily accomplished by \textit {htseq-counts}  by Sanders et al  2014 \cite{Sanders2014} or \textit{featureCounts}  [ R package \textbf{Rsubread} ] by Liao et al 2013 \cite{Liao2013}.  RNA-seq raw counts are the basis of all statistical workflows, be it exploration or differential expression analysis [\textbf{edgeR} \cite{Robinson2010}, \textbf{limma} \cite{Ritchie2015} ]. There is a growing trend to make the analysis ready raw counts tables openly accessible for statistical analysis. ReCount is a online site that hosts RNA-seq gene counts datasets from 18 different studies \cite{Frazee2011} along with relevant metadata. Such gene counts datasets are the inputs for our clustering algorithm. \\[2 pt]
%

% In the preprocessing step before applying our method, we remove the genes with 0 or same count of matched reads across all samples (non-informative genes), any sample or gene  with NA values of reads and  ERCC spike-in controls,  as the latter may create bias due to their typical very high expression (number of reads mapped to them).  For illustration, we applied our method GTEx Version 6 tissue level gene counts data \cite{GTEX2013} and on a couple of single cell data due to Zeisel \textit{et al} \cite{Zeisel2015} and Jaitin \textit{et al} \cite{Jaitin2014}. 
 
% If $C_{ng}$ is the gene count for $g$ th gene in tissue sample $n$, then we define the thinned counts as 
%
%$$ c_{ng}  \sim Bin(C_{ng}, p_{thin} )  $$
%
%where $p_{thin}$ is the thinning probability. W chose $p_{thin}$ to be of the order of the ratio of the total number of reads mapped to a single cell experiment (in this case Zeisel et al (2015) data for instance) and the total number of reads in the GTEx dataset, which turned out to be approximately 0.0001. To check for robustness of our clustering algorithm, we varied $p_{thin}$ to be $0.01, 0.001, 0.0001$ (see Fig ).  
%

\subsection{Model overview}

The method we adopt allows each sample to have a grade of membership in the underlying clusters. This type of approach was first used in population genetics to model population samples as having parts of its genome derived from different clusters that represented ancestries (see Pritchard, Stephens and Donnelly 2000 \cite{Pritchard2000}). Later, a similar approach was used to model documents to have grades of membership in different clusters, representing topics, depending on the frequencies of different words used (see Blei, Ng and Jordan 2003 \cite{Blei2003}). In RNA-sequencing data too, each sample is viewed as having grades of membership in different underlying subpopulations (for e.g. cell types for tissue samples, cell phases for single cells etc). 

We assume that the row vector of counts for each sample $n$ across the genes follows a multinomial distribution.

$$ c_{n*} \sim Mult(c_{n..}, p_{n*}) $$

where $c_{n*}$ is the count vector for the $n$ th sample, $c_{n..}$ is the sum of the counts in the vector $c_{n*}$, and $p_{n*}$ is the probability that a read coming from sample $n$ would get assigned to one of the $G$ genes. The main idea is that this read could come from hidden subpopulations (may be cell types for tissue level expression study or cell cycle phases for single cell study) and its probability of getting assigned to some gene $g$ may depend on which subpopulation it comes from. Denote  the probability that the sample is coming from the $k$ th subpopulation by $q_{nk}$ ($q_{nk} \geq 0$ and $\sum_{k=1}^{K} q_{nk} =1$ for each $n$).  Given that the sample is coming from the $k$th subgroup, the probability of a read being matched to the $g$th gene is given by $\theta_{kg}$ ($\theta_{kg} \geq 0$ and $\sum_{g=1}^{G} \theta_{kg} =1$ for $k$th subgroup). Then one can write 

$$ p_{ng} = \sum_{k=1}^{K} q_{nk}\theta_{kg}   \hspace{1 cm}  \sum_{k=1}^{K} q_{nk}=1 \hspace{1 cm} \sum_{g=1}^{G} \theta_{kg}=1 $$

This model has $N \times (K-1) + K \times (G-1)$ many unconstrained parameters, which is much smaller than the $N \times G$  data values of counts. Usually for RNA-seq samples $N$ varies in the region of $100$s to $1000$s  and $G$ ranges from $10,000$ to $50,000$ (depending on the underlying species and the types of genes tokenized) and $K << \{N,G \}$. 

Due to the widespread use of topic models in Natural Language Processing literature, many softwares are available for fitting such models on document collections. Here, we use a Maximum a posteriori (MAP)  estimation procedure implemented in R package  \textbf{maptpx}(see Taddy 2012 \cite{Taddy2012}).

% It assumes the priors
%
%$$ q_{n*} \sim Dir ( \frac{1}{K}, \frac{1}{K}, \cdots, \frac{1}{K} ) $$
%$$ \theta_{k*} \sim Dir(\frac{1}{KG}, \frac{1}{KG}, \cdots, \frac{1}{KG} ) $$
%
%For better estimation stability, the usual parameters of the model are converted to natural exponential family parameters to which one can apply the EM algorithm ). The value of the Bayes factor for the model with $K$ clusters compared to the model with 1 cluster, is recorded for each $K$, and the optimal $K$ is chosen by running the clustering method for different choices of $K$ and then choosing the one with maximum Bayes factor. The two main outputs from this method are the $Q_{N \times K}$ topic proportion matrix  and $F_{K \times G}$ relative gene expression for each cluster.

\subsection{Visualization}

A nice way to visualize the amount of relatedness among the samples is through the Structure plot, which is a popular tool to visualize the admixture patterns in population genetics based on SNP/ microsatellite data (see Rosenberg \textit{et al} 2002 \cite{Rosenberg2002}).  
For each sample $n$, $q_{nk}$  represents the relative abundance of  hidden subgroup $k$ in the sample. The Structure plot  assigns a color to each of the subgroups and then presents a vertical stacked barchart for each individual, with stack heights representing subgroup proportions and each subgroup assigned a particular color. If the colored patterns of two bars are similar, then the two samples must be closely related.  

Another visualizing tool we recommend is t-distributed Stochastic Neighbor Embedding (t-SNE), which is well-suited for visualizing the high dimensional datasets on 2D, preserving the relative distance between samples in high dimension to a fair extent in 2D (see L.J.P. van der Maaten \cite{Maaten2014} and L.J.P. van der Maaten and Hinton \cite{Maaten2008}). t-SNE shows us which samples are close to each other when the data is projected on 2D. t-SNE is not a clustering tool and unlike Structure plot, does not show the relative abundance patterns of different subgroups in the sample. However, both Structure plot and t-SNE give a lot more interpretable visualization of the clustering compared to the heatmap and hierarchical clustering (see Results for illustration).

\subsection{Cluster annotation}

A question of considerable biological interest is which genes are significantly differentially expressed across the clusters, or in other words, which genes are driving the clustering. To answer this, we fix one cluster and for each gene, define a distance metric between that cluster and any of the other clusters, based on the cluster expression profile of the gene, namely the $\theta$ values. For cluster $k$, we define the distance from cluster $l$ based on expression profile of gene $g$ to be 

$$ KL^{g} [k,l] : = \theta_{kg} \; log \frac{\theta_{kg}}{\theta_{lg}} + \theta_{lg} - \theta_{kg} $$

This is similar to the Kullback Leibler divergence of the Poisson distribution with parameter $ \theta_{lg}$ from another Poisson distribution with parameter $ \theta_{kg}$. For each cluster $k$, we define the divergence measure for gene $g$ as 

$$ Div^{g}[k] = \underset{l \neq k}{min} \; KL^{g} [k, l] $$

The higher the divergence measure, the more significant is the role of the gene in the clustering. We choose a small subset of around 50-100 genes with highest values of $Div^{g}[k]$ for each $k$ and this set of genes can be viewed to be the most important genes driving the cluster $k$.  \\[1 pt]

Once the most important driving genes for each cluster $k$ have been extracted, we perform gene annotations on them using \textbf{mygene} R Bioconductor package (due to Mark A, Thompson R and Wu C  2014 \cite{Thompson2014}). We check if the driving genes for a particular cluster are associated with some specific biological functionality. This would validate whether the subgroups are biologically relevant. 

%For instance, for GTEx tissue sample data, if the clusters are indeed driven by cell types, then the top genes for these clusters will probably be associated with proteins related to  functions for that particular cell type.
%

%we fix each gene and then look at the KL divergence matrix of one cluster/subgroup $k$ relative to other cluster/subgroup $k^{'}$, which we call $KL^{g}_{K \times K}$. This matrix is symmetric and has all diagonal elements $0$ as the divergence of a cluster with respect to itself is $0$. 




%Then we perform gene annotations for the top genes in each subgroup 

\medskip
















 









