\title{\Large{\textbf{Model-based clustering of RNA-seq expression data using grade of membership models}}}
\author{ Kushal K Dey  \qquad Matthew Stephens}

\maketitle

\begin{abstract}
We argue for the benefits of model-based approach to clustering expression data from bulk RNA and single cell RNA-seq data. It is similar to the topic model in Natural Language Processing and the admixture model in Population Genetics, that assigns grades of cluster membership to each sample. This approach provides us with easily interpretable cluster visualization  and  detects the underlying structure in the data better than distance based approaches. It also extracts the important genes that drive the clusters and provides measures of model fit to assess the strength of clustering. Further, we show that this method is robust under low coverage of reads. We apply this method on the GTEx tissue level bulk RNA expression data as well as two single cell RNA-seq data. Our methods are implemented in a R package \textbf{CountClust}, available at \url{https://github.com/kkdey/CountClust}.
\end{abstract}

\section{Introduction}

Ever since large-scale gene expression measurements have been possible using micro-arrays, clustering has played a major role in their analysis \cite{Eisen1998}.
For example, clustering can aid in quality control, and helps elucidate heterogeneity among samples. However, although gene expression 
measurement technologies have moved on -- with gene expression measurements now routinely made with sequencing technologies (RNA-seq), sometimes at the resolution of single cells (sc-RNA-seq) -- clustering methods used have remained largely the same. Indeed, the most commonly-used clustering techniques in gene expression studies
remain the distance-based hierarchical clustering 
methods used in early microarray work \cite{Eisen1998}. This contrasts with many other fields, where model-based clustering methods
have become widely used, and in many cases the method of choice (e.g.~\cite{Pritchard2000}). Our goal here
is to argue that such model-based approaches also provide an attractive approach to cluster analysis of RNA-seq data, both bulk and single-cell.
In particular we illustrate the potential for ``grade of membership models" \cite{Erosheva2006} to elucidate structure in both bulk and single-cell RNA-seq expression data. 

Grade of membership models are generalizations of clustering models that allow each sample to have partial membership in more that one cluster.
That is, they allow that each sample has a proportion, or ``grade" of membership in each cluster. Such
models are widely used in population genetics to model ``admixture", where individuals can have ancestry from multiple populations \cite{Pritchard2000},
and in document clustering (\cite{Blei2003,Blei2009}) where each document can have membership in multiple ``topics". In these fields
the grade of membership models are often known as ``admixture models", and ``topics models" or ``Latent Dirichlet Allocation" \cite{Blei2003}.

Here we apply grade of membership models to RNA-seq expression data. In this context these models allow that each
 sample has some proportion of its RNA-seq reads coming from each cluster. For bulk RNA-seq experiments this assumption 
could be motivated by a simple -- or perhaps simplistic -- argument: each sample is a mixture of different cell types, and so clusters 
could represent cell types, and the membership of a sample in each cluster could represent the proportions of each cell type present.
This is similar to the idea of ``deconvolution" methods that use cell-type-specific expression profiles of marker genes to estimate the concentration of different cell types in a mixture \cite{}. And, indeed, the grade of membership model we use here is analogous to blind deconvolution approaches \cite{Schwartz2010,Repsilber2010}
 which estimate cell type proportions and cell type signatures jointly (see also \cite{Shen-Orr2010,Qiao2012} for semi-supervised approaches), although 
 our models differ from these previous works in the use of a multinomial model for the RNA-seq read counts. 
However, we believe that the grade of membership model can be useful more generally to elucidate structure in expression data.
For example, in single-cell expression data treating each sample as a ``mixture of cell types" is clearly inappropriate, and yet we see value in the idea
that there may be some ``continuous" variation in cell types, rather than (or perhaps in addition to) the purely discrete variation captured by cluster models. 
Indeed, the extent to which variation among cells can be described in terms of discrete clusters vs more continuous populations
seems a fundamental question that, when combined with appropriate single-cell RNA-seq data, the grade of membership models used here may
ultimately help address. Further, even for bulk RNA-seq data, we argue that grade of membership models may yield interesting insights into heterogeneity among samples
even if the inferred cluster membership do not correspond precisely to proportions of specific cell types, as may often happen in practice.

Interestingly, several software packages capable of fitting these grade of membership models to RNA-seq data are already available! This is because
the models used in natural language processing for clustering documents based on their word counts are multinomial models that
apply naturally and immediately to RNA-seq data. Whereas documents are characterized by counts of each possible word in a dictionary, RNA-seq samples
are characterized by counts of reads mapping to each possible gene (or other unit, such as transcript, or exon) in the genome. 
Here we use the R package {\tt maptpx} \cite{Taddy2012} to fit these models, and we add functionality for visualizing the results and annotating
clusters by their most distinctive genes to help biological interpretation. These methods are implemented in the R package {\tt count-clust} available
from \url{}.  



%In this paper, we demonstrate that for RNA-seq (bulk or single cell) data with known structural patterns, such count clustering approach identifies the structure better than hierarchical clustering. It also allows one to interpret each cluster by providing information about genes that are playing a significant role in driving the clusters and these genes may be important from both biological and medical standpoint. Also we show our method to be robust even for low coverage data as might be the case for single cell RNA-seq (scRNA-seq) data.
%We illustrate the performance of our method on GTEx tissue level  bulk-RNA seq data as well as on two single cell data (due to Jaitin \textit{et al} 2014 \cite{Jaitin2014} and Deng \textit{et al} 2014 \cite{Deng2014}) . 


%text from methods


%The main idea is that this read could come from hidden subpopulations (may be cell types for tissue level expression study or cell cycle phases for single cell study) and its probability of getting assigned to some gene $g$ may depend on which subpopulation it comes from. Denote  the probability that the sample is coming from the $k$ th subpopulation by $q_{nk}$ ($q_{nk} \geq 0$ and $\sum_{k=1}^{K} q_{nk} =1$ for each $n$).  Given that the sample is coming from the $k$th subgroup, the probability of a read being matched to the $g$th gene is given by $\theta_{kg}$ ($\theta_{kg} \geq 0$ and $\sum_{g=1}^{G} \theta_{kg} =1$ for $k$th subgroup). Then one can write 



%\begin{itemize}
%
%\item \textit{objectives of the work}: to devise a completely unsupervised method to cluster the samples (tissue or single cell samples) into biologically meaningful sub-types based on the RNA-seq gene counts data
%
%\item \textit{justification of objectives} : 
%\begin{enumerate}
%
%\item  People have mainly used hierarchical clustering from GTEx consortium paper to most single cell RNA seq papers I have come across. We have evidence Admixture model does better than hierarchical clustering from  a biological viewpoint ( see structure.beats.hierarchical.html).
%
%\item Hierarchical clustering does not give us directly the genes that drive the clusters, Admixture model does, and it also provides us with a log likelihood to fix how many clusters to choose, based on Bayes factor. 
%
%\item We can predict the admixture proportions of cell types in any new sample coming in, so we can easily cluster new samples in cancer biopsy where the sub-types may involve cancer or non-cancer samples.
%
% 
%
%\end{enumerate}
%
%\item \textit{Background}
%\begin{enumerate}
%\item The BackSpin algorithm used by Zeisel et al. Claim is it does better than hierarchical but not model based (also not convincingly proven to be better)
%
%\item Use of downsampling and then modified hierarchical clustering scheme as applied by Jaitin et al.
%
%\item Mainly, people have used hierarchical clustering scheme
%
%\item Population genetics uses Admixture model on a regular basis. We think we can generalize that to RNA-seq data. The only question is do we really see the tissue samples as cell type admixture, as we observe individuals as population admixture. The answer seems to be yes.
%
%\end{enumerate}
%
%\item \textit{Guidance to the reader}
%\begin{enumerate}
%\item The Structure plot and t-SNE plots  for GTEx tissues and for Zeisel data. Much better visualization than the regular heatmaps that we tend to see in RNA-seq papers. 
%
%\item The Structure plot analysis for Brain samples that shows $80\%$ one cluster in cerebellum tissue samples and then from gene annotations, it is revealed this  cluster is indeed associated with synaptic activities implying it must be neuronal cell types. This is pretty cool because we have a priori knowledge from cell type specific markers that around $80\%$ of cells in cerebellum are neurons.
%
%\item Also the strategy is similar to the topic model strategy in natural language processing and it is a really nice technique to use for RNA-seq datasets clustering.
%
%\end{enumerate}
%
%\end{itemize}
%
