\section{Results}

%An outline for results  (under consideration)
%
%\begin{itemize}
%
%\item Form two separate subsections, one for the GTEx Version 4 data and the other for the single cell Zeisel data.
%
%\item For GTEx data, give a figure comprising of 4 Structure plots for different $K$s, may be $2,5,10,15$. Fix the thinning parameter $p_{thin}$ to say $0.0001$.  Also record the log likelihoods (Bayes factors) for each of the 4 models, as reported by \textbf{maptpx}. 
%
%\item Have one figure showing the robustness of the clustering method on the thinning parameter $p_{thin}$. Fix $k=10$ and vary $p_{thin}$ to be $0.0001$, $0.001$ and $0.01$. 
%
%\item One t-SNE plot for GTEx samples (with and without admixture in the same plot). Should this be in results or in discussions? Also the t-SNE probably would require an electronic supplemental file as I would need the \textbf{qtlcharts} highlighting for those plots. 
%
%\item The GTEX brain samples Structure plot  for $K=4$ that shows the neuron cell types in brain cerebellum and cerebellar hemisphere. That is to show that the clusters are driven by cell types.
%
%\item Gene annotations for the GTEx significant genes (for brain) and also for the general set up (to decide on which $K$ to fix). Use Bayes Factor?
%
%\item The Structure plot for Zeisel single cell data. again Multiple $K=2,5,7,10$. 
%
%\item Gene annotations for the Zeisel single cell data. Need to choose the optimal $K$. Use Bayes Factor?
%
%\item t-SNE plot of the admixture proportions??..Is that required? Depends on how we present t-SNE. If this goes to discussion, we will avoid it here
%
%\end{itemize}

\subsection{Clustering human tissues based on bulk RNA expression}

We begin by illustrating the GoM model on bulk RNA expression measurements from the GTEx project (V6 dbGaP accession phs000424.v6.p1, release date: Oct 19, 2015, \url{http://www.gtexportal.org/home/}).  These data consist of per-gene read counts from RNA-seq performed on $8555$ samples collected from $450$ human donors across $51$ tissues and LCL and transformed fibroblast cell-lines. We analyzed $16,069$ genes that satisfied filters (e.g. exceeding certain minimum expression levels) that were used during eQTL analyses by the GTEX project (gene list available in \url{https://github.com/stephenslab/count-clustering/blob/master/utilities/gene_names_GTEX_V6.txt}). 

To assess structure in these data we applied the grade-of-membership model with $K=10,12,15$. Although results differ with $K$, many of the primary patterns were consistent across $K$. Here, for brevity, we focus on results for $K=15$, shown as a Structure plot in \textbf{Figure \ref{fig:fig1}(a)} (see also an alternative visualization using a 2-dimensional projection with t-sne \cite{Maaten2008}, \cite{Maaten2014}, in \textbf{Supplementary Fig 1 \url{http://stephenslab.github.io/count-clustering/project/src/tissues_tSNE.html}}). Reassuringly, much of the structure highlighted by these results follows the known division of samples into tissues: that is, samples from the same tissue tend to have similar grades of membership across clusters. Some tissues are represented by essentially a single cluster (e.g.~Pancreas, Whole Blood), whereas other tissues are represented as a mixture of multiple clusters (e.g.~Thyroid). Furthermore, the results highlight biological similarity among some tissues by assigning samples from those tissues similar membership proportions. For example, samples from different parts of the brain have similar memberships, as do the arteries (Artery-aorta, Artery-tibial and Artery-coronary) and skin (Skin Not Sun Exposed- suprapubic and Skin Sun Exposed- lower leg). Samples from the tibial nerve have small but consistent amounts of membership in common with brain tissues, as well as larger amounts in common with the adipose tissues (Adipose Subcutaneous and Adipose Visceral (Omentum)). Indeed, many 
tissues show membership in the red ``Adipose" cluster, possibly reflecting, at least in some cases, contamination with adipose cells.
 
Each cluster in our model is characterized by a vector that contains the mean expression level for each gene. To help biologically interpret each cluster we annotate it by identifying the genes whose expression levels most strongly distinguish that cluster from the others (see Cluster Annotation, in Methods and Materials).  \textbf{Tab ~\ref{tab:tab1}} summarizes the results of this cluster annotation (top three genes) for the GTEX analysis in Figure \ref{fig:fig1}a.  Again, reassuringly, the results consistently align with known biology. For example,  the top three genes driving the light brown cluster, which distinguishes Pancreas from other tissues, are  \textit{PRSS1}(protease serine 1), \textit{CPA1} (carboxypeptidase) and \textit{PNLIP} (pancreatic lipase), all of which are intimately involved in pancreatic function. Similarly, the top three genes driving the grey cluster, which distinguishes Whole Blood, are all hemoglobin genes, \textit{HBB} (hemoglobin, beta), \textit{HBA2} (hemoglobin, alpha 2) and \textit{HBA1} (hemoglobin, alpha 1).  Similarly, spermatogenesis and sperm-related genes characterize the Testis cluster, Keratin-related genes characterize the skin cluster, Myosin-related genes characterize the muscle skeletal cluster, etc. In cases where a cluster occurs in multiple tissues these annotations may be particularly helpful for understanding what may be driving this co-membership. For example, the top three genes in the  burlywood cluster  which is common to Lung, Spleen and Small Intestine - Terminal Ileum, code for surfactant proteins B, A2 and A1.


%A field of very active interest in recent times is to estimate the proportion of different cell types in different tissues. Marker based approaches are usually adopted to validate for different cell types and get a sense of the abundance of different cell types in the tissue samples \cite{Grun2015} \cite{Palmer2005}. The admixture model is a marker free method to obtain clusters driven by cell types. 

Although global analysis of all tissues is useful for highlighting major structure in the data, it may be less effective at identifying finer-scale structure within tissues or among similar tissues. For example, here our global analysis allocated similar cluster memberships to all brain tissues,  and we suspected that these tissues may exhibit substructure that could be uncovered by analyzing the brain samples separately.  \textbf{Fig \ref{fig:fig1}(b)} shows the Structure plot for $K=4$ on only the Brain samples. The results highlight much finer-scale structure compared with the global analysis. Brain Cerebellum and Cerebellar hemisphere are essentially assigned to a separate cluster, whose top 3 defining genes are SNAP25 (synaptosomal-associated protein, 25kDa), ENO2 (enolase 2- gamma, neuronal) and CHGB (chromogranin B),  all of which are associated with neuronal activities (\textbf{Supplementary Table 1}). %[POssibly GRANULE CELLS?]
The spinal cord samples also show consistently strong membership in a single cluster, whose top 3 defining genes are MBP  (involved in myelination \cite{Hu2016}) and MYH11 and ACTA2, both of which are .. Genetic mutations in the latter two genes can cause TAAD Thoracic aortic aneurysms/dissection \cite{Renard2013}. The remaining samples all show membership in multiple cluster, with cortex samples being distinguished from other samples by stronger membership in a cluster (blue in figure) whose top 3 defining genes include UCHL1, which is expressed in neurons and is required for normal synaptic and cognitive function \cite{Gong2006}.


% It seems the red cluster is mainly prominent in what are called sub-cortical regions of the brain (which are immediately below the cortex).  In that sense the four clusters, although not fully representative, do seem to be driven by the cerebellum, cortex, sub-cortex and spinal cord. These are anatomically meaningful and also functionally.  Functionally, sub-cortical regions of the brain are associated with lower order thinking tasks and these are regions that are present in reptiles and birds as well. Cortex is mainly present in mammals and is used for higher order thinking tasks. Cerebellum main function is in motion learning. Also amygdala and hippocampus are anatomically very close to each other and constitute the limbic region of the brain \url{https://s3.amazonaws.com/classconnection/118/flashcards/3700118/png/screen_shot_2012-09-17_at_75637_am1347883157040-1517F5218E616F44DE2.png}. Even substantia nigra is pretty close to both of them. Also cerebellar granule cells do constitute bulk of the cerebellum and are the principal granule cells in human body (WIKI). Given that the genes we obtained as driving genes for the clusters seem to be associated with cerebellar granule cells, seems it is meaningful to call the green cluster to be driven by the granule cells. Myelin basic protein has been found to have higher expression in spinal cord cluster when compared to other clusters. Myelin increases nervous transmission and nervous transmission is very swift through the spinal cord. This could explain why MBP is one of the top distinguishing genes for the spinal cord cluster. 




%Recent stereological approaches have shown that rat cerebellum contains $> 80 \%$ neurons (Herculano-Houzel and Lent 2005) \cite{Houzel2005}, much higher than other parts of the brain. 

\subsection{Quantitative comparison with hierarchical clustering}

Since most clustering of gene expression datasets is performed using distance-based hierarchical clustering methods, we compared the accuracy of these distance-based methods with the model-based approach. Specifically, for each pair of tissues in the GTEX data we assessed whether or not each clustering method
correctly partitioned samples into the two groups (See methods for details.) Overall the model-based method succeeded in $59 \%$ of the comparisons, compared with $29 \%$ for the distance-based method; Figure \ref{fig:fig2}.

\subsection{Robustness to sequencing depth}

RNA-sequencing of single-cell data typically involve substantially lower effective sequencing depth compared with bulk experiments, due to the lower number of molecules available to sequence in a single cell.
Therefore, we checked robustness of our results above to sequencing depth. Specifically we repeated analyses using thinned data designed to mimic lower sequencing depth: if $c_{ng}$ is the counts of number of reads mapping to gene $g$ for sample $n$ for the original data, then the thinned counts are given by 
\begin{equation}
t_{ng} \sim Bin (c_{ng}, p_{thin})
\end{equation}
where we used $p_{thin} =0.0001$. With this value of $p_{thin}$ total sequencing depth is similar to the per-sample depth in the 
single-cell data from \cite{Jaitin2014} considered below. (See  \textbf{Fig \ref{fig:figS3}} [check!] and \textbf{Fig \ref{fig:figS2}} for results with other values of $p_{thin}$.) 

For the thinned GTEX data the Structure plot for $K=15$ (\textbf{Fig \ref{fig:fig1}(c)}) preserves most of the major features of the original analysis on unthinned data (\textbf{Fig \ref{fig:fig1}(a)}). For the comparisons with distance-based methods, both methods suffer reduced accuracy for the thinned data, but the model-based method retains its superior performance (xx vs yy).  

 %Then we generated a set of $50$ samples randomly drawn from the pooled set of samples coming from these two tissues and then observed whether the hierarchical and the admixture were separating out samples coming from the two different tissues.


\subsection{Clustering of single-cell RNA-seq data}

To examine the potential for model-based clustering of single cell RNA-seq data we examined two datasets, from Jaitin \textit{et al} \cite{Jaitin2014} and Deng \textit{et al} \cite{Deng2014}.  

Jaitin \textit{et al} sequenced over $4000$ single cells from mouse spleen. Following the original authors protocol, we also filtered out 16 genes that they found to show significant batch-specific expression. Here we analyze 1041 of these cells that were categorized as $CD11c+$ in the \textit{sorting markers} column of their data (\url{http://compgenomics.weizmann.ac.il/tanay/?page_id=519}), and which had total number of reads mapping to non-ERCC genes greater than $600$. (We believe these cells correspond roughly to the 1040 cells in their Figure S7.)   Our hope was that applying our method to these data would identify, and perhaps refine, the cluster structure evident in Figures 2A and 2B of \cite{Jaitin2014}. However, our method yielded rather different results (Figure \ref{fig:fig3}). Each cell was assigned to multiple clusters, and the cluster membership vectors strongly correlated with amplification batch (which in turn is strongly correlated with sequencing batch). The fact that batch effects are detectable in data like these is not particularly surprising. There is a growing recognition of the importance of batch effects in high-throughput data generally \cite{Leek2010} and in single cell data specifically \cite{Hicks2015}. And indeed, dimension reduction methods such as the ones we use here can be helpful in controlling for such effects \cite{Leek2007} \cite{Stegle2012}. However, why these batch effects are not visible in the original analyses \cite{Jaitin2014}, is unclear. 



%identify distinct clusters, corresponding to the clusters of B cells, NK cells, pDCs and monocytes
%focusing initially on a heterogeneous mix enriched for expression of the CD11c surface marker. 
% The aim of their study was to separate out the B cells, NK cells, pDCs and monocytes.   \textbf{Fig \ref{fig:fig3}} (\textit{top panel}) presents the Structure plot  for $K=7$ for the Jaitin \textit{et al} data with the samples arranged by their amplification batch (which was a refinement of the sequencing batch). This highlights the need for caution regarding interpreting Admixture results or any clustering results, as there is a possibility of technical effects driving the clusters instead of true biological effects. There has been a growing concern among biostatisticians today about how to deal with batch effects \cite{Leek2010} \cite{Hicks2015}. \\[1 pt]

Deng \textit{et al} collected expression data from individual cells from zygote to blastocyst stages of mouse preimplantation development \cite{Deng2014}. Deng \textit{et al}'s analysis focussed particularly on allele-specific expression from the two contributing mouse strains (CAST/EiJ and C57BL/6J). Here we analyze the counts of the two alleles combined. Visual inspection of the Principal Components Analysis in \cite{Deng2014} suggested 6-7 clusters, so we fit the cluster model with $K=6$. 
The results (Figure \ref{fig:fig4}) clearly highlight the structure in the different development stages starting from zygote, through early/mid/late 2 cells, 4 cells, 8 cells, 16 cells, and early/mid blastocyst to finally late blastocyst. Specifically, cells that are from the same stage show similar cluster membership proportions. Further, many of the clusters show notable trends through the stages. For example, 
membership in the red cluster is non-existent in early stages, starts in the 4-cell stage, becomes more prominent in the 8-16 cell stages, drops substantially in the early and mid-blastocyte stages, and is essentially absent in the late blastocytes. More generally, cluster memberships for cells from adjacent stages tend to be more similar to one another than those for cells from distant stages. 

Examining the clustering results by embryo highlights apparent embryo-level effects in the early stages (Figure \ref{fig:fig4}): that is, cells from the same embryo sometimes showed distinctive differences from other embryos. For example, the two cells from one of the 2-cell embryos (check) shows much stronger membership in the magenta cluster than other 2-cell embryos, and four cells from one of the 4-cell embryos (embryo 4) shows consistently more yellow membership than the other 4-cell embryos. 

Finally, the results indicate a few samples that appear to be outliers - for example, a cell from a 16-cell embryo that looks like a very early stage cell (zygote or early 2-cell), and a cell from an 8-stage embryo that looks rather different from any of the others.

Notably, for both these single-cell data sets, most cells are assigned to a combination of more than one cluster, rather than a single cluster (the exception being the very early-stage cells in data from Deng et al). This highlights the potential utility for grade-of-membership models to capture structure in single cell data that might be missed by simpler cluster-based approaches.



%, 
%We find that admixture model is more successful in separating out different tissues in general, compared to the hierarchical clustering technique. The admixture model is essentially a count based modeling approach and seems to handle low counts and zero counts much better than the hierarchical method which is a more general approach to clustering. Since the RNA-seq data and in particular scRNA-seq data have lots of low counts and zero counts, the admixture model seems to be more suited for such data compared to hierarchical clustering method. 

%Currently there is a lot of interest in single cell sequencing as it is more informative about individual cell expression profiles compared to the RNA-seq on tissue samples. We were curious to see how stable the Admixture results are if the GTEx RNA-seq data is viewed at the scale of a single cell data. We achieve the latter by thinning the GTEx data under thinning parameter $p_{thin}=0.0001$ which is the order of scale obtained by dividing the total library size of the Jaitin \textit{et al} \cite{Jaitin2014} with respect to the library size of the GTEx V4 read counts data. We fitted the admixture model for $K=12$ on the thinned data and the Structure plot for the fitted model is presented in \textbf{Fig \ref{fig:fig4}}. It seems that most of the features observed in \textbf{Fig \ref{fig:fig1}} seem to be retained, for instance- the Brain samples clustering together, Whole blood and Testis forming separate clusters, Muscle skeletal and Heart tissue samples showing very similar patterns etc. However, thinning indeed shrinks the small differences across tissues and makes it more difficult to distinguish between tissues, as evident from the comparative study of hierarchical and admixture models, analogous to \textbf{Fig \ref{fig:fig3}}, for thinned data with thinning parameters $p_{thin}=0.001$ and $p_{thin}=0.0001$ in  \textbf{Fig \ref{fig:figS2}}. One can see that with thinning, the performance of admixture model in separating the tissues deteriorates but encouragingly, it seems that admixture does outperform the hierarchical clustering even under thinned data. \\[3pt]

 








