\section{Methods and Materials}

\subsection{Data preprocessing}

RNA-seq experiments usually provide us with a set of FASTQ files that contain the nucleotide sequence of each read and a quality score at each position, which can be mapped to  reference genome or exome or transcriptome. The output of this mapping is usually saved in a SAM/BAM file using SAMtools  \cite{Li2009} . This task is primarily accomplished by \textit {htseq-counts}  by Sanders et al  2014 \cite{Sanders2014} or \textit{featureCounts}  [ R package \textbf{Rsubread} ] by Liao et al 2013 \cite{Liao2013}.  RNA-seq raw counts are the basis of all statistical workflows, be it exploration or differential expression analysis [\textbf{edgeR} \cite{Robinson2010}, \textbf{limma} \cite{Ritchie2015} ]. There is a growing trend to make the analysis ready raw counts tables openly accessible for statistical analysis. ReCount is a online site that hosts RNA-seq gene counts datasets from 18 different studies \cite{Frazee2011} along with relevant metadata. We start with such gene count datasets and assume that we have samples (say $N$)  along the rows and the genes  (say $G$) along the columns. Before we apply our methods, we remove the genes with 0 count of matched reads across all samples, implying that these genes are probably not expressed in any sample and hence non-informative for the clustering or differential analysis of the samples. We also remove the samples or genes with NA values of reads, if any. Additionally we also remove any ERCC spike-in controls as they may create bias to the biological clustering patterns.  For illustration, we have applied our method on a single-cell RNA seq data  due to Zeisel et al (2015) \cite{Zeisel2015} and GTEx Version 4 gene counts data \cite{GTEX2013}. The GTEx data is a tissue sample data and the reads are recorded for multitude of cells present in the tissue sample. This can lead to really large values of read counts, in particular for highly expressed genes. To reduce the model over-dispersion and to make the analysis comparable to single cell datasets, we applied a thinning  mechanism to the GTEx data. If $C_{ng}$ is the gene count for $g$ th gene in tissue sample $n$, then we define the thinned counts as 

$$ c_{ng}  \sim Bin(C_{ng}, p_{thin} )  $$

where $p_{thin}$ is the thinning probability. W chose $p_{thin}$ to be of the order of the ratio of the total number of reads mapped to a single cell experiment (in this case Zeisel et al (2015) data for instance) and the total number of reads in the GTEx dataset, which turned out to be approximately 0.0001. To check for robustness of our clustering algorithm, we varied $p_{thin}$ to be $0.01, 0.001, 0.0001$ (see Fig ).  

\subsection{Methods Overview}

We use a topic model approach due to Matt Taddy (package \textbf{maptpx}) to perform the clustering of the samples based on RNA-seq reads data \cite{Taddy2012}. We denote this matrix of counts by $C_{N \times G}$ where $N$ is the total number of samples (tissue/single cell) and $G$ is the number of genes.  We assume that the row vector of counts for each sample $n$ across the genes is multinomially distributed. 

$$ c_{n*} \sim Mult(c_{n..}, p_{n*}) $$

where $c_{n*}$ is the count vector for the $n$ th sample, $c_{n..}$ is the sum of the counts in the vector $c_{n*}$, and $p_{n*}$ is the probability that a read coming from sample $n$ would get assigned to one of the $G$ genes. \\[2 pt]
The idea here is that this read could be coming from some cell type for the tissue level expression study (or from some cell cycle phase for the single cell case study) and its probability of getting assigned to some gene $g$ will depend on which cell type (cell cycle phase) it comes from. In general, we may assume that the read is coming from one of the several (say $K$) underlying classes/groups, which are not observed. Denote  the probability that the sample is coming from the $k$ th subgroup by $q_{nk}$ ($q_{nk} \geq 0$ and $\sum_{k=1}^{K} q_{nk} =1$ for each $n$) and the probability of a read coming from the $k$th subgroup, to be matched to the $g$th gene, by $\theta_{kg}$ ($\theta_{kg} \geq 0$ and $\sum_{g=1}^{G} \theta_{kg} =1$ for $k$th subgroup). Then one can write 

$$ p_{ng} = \sum_{k=1}^{K} q_{nk}\theta_{kg}   \hspace{1 cm}  \sum_{k=1}^{K} q_{nk}=1 \hspace{1 cm} \sum_{g=1}^{G} \theta_{kg}=1 $$

This model has in all $N \times (K-1) + K \times (G-1)$ many unconstrained parameters, which is much smaller than the $NG$ many counts data we have. Usually $K << min \{N,G \} $ and for RNA-seq datasets, $N$ is usually in the region of $100$s to $1000$s  and $G$ range from $20,000$ to $50,000$.  To estimate the model, a Maximum a posteriori (MAP) based approach is used. It assumes the priors

$$ q_{n*} \sim Dir ( \frac{1}{K}, \frac{1}{K}, \cdots, \frac{1}{K} ) $$
$$ \theta_{k*} \sim Dir(\frac{1}{KG}, \frac{1}{KG}, \cdots, \frac{1}{KG} ) $$

For better estimation stability, the usual parameters of the model are converted to natural exponential family parameters to which one can apply the EM algorithm (see Taddy 2012 \cite{Taddy2012}). The value of the Bayes factor for the model with $K$ clusters compared to the model with 1 cluster, is recorded for each $K$, and the optimal $K$ is chosen by running the clustering method for different choices of $K$ and then choosing the one with maximum Bayes factor. The two main outputs from this method are the $Q_{N \times K}$ topic proportion matrix  and $F_{K \times G}$ relative gene expression for each cluster.

\subsection{Post processing analysis}

For each $n$, $q_{nk}$'s which will give an idea about the relative abundance of individual subgroups (cell functional groups or cell types) represented in the sample (single cell or tissue respectively). If two samples $n$ and $n^{'}$ are very close, say both coming from the same tissue for the tissue level data, then we expect $q_{n*}$ and $q_{n^{'}*}$ to be very close too. A nice way to visualize the amount of relatedness among the samples is through the Structure plot due to Pritchard Lab, which is a popular tool to visualize the admixture patterns in population genetics based on SNP/ microsatellite data \cite{Pritchard2000} \cite{Raj2014}. The Structure plot  assigns a color to each of the subgroups and then presents a vertical barplot for each individual, which is fragmented by the subgroup proportions and colored accordingly. If the colored patterns of two bars are similar, then the two samples must be closely related. The other visualizing tool we use is t-distributed Stochastic Neighbor Embedding (t-SNE) due to Laurens van der Maaten, which is well-suited for visualizing the high dimensional datasets on 2D, preserving the relative distance between samples in high dimension to a fair extent in 2D \cite{Maaten2008} \cite{Maaten2014}.  

The other question of interest is which genes are significantly differentially expressed across the clusters, or in other words, which genes are driving the clustering. To answer this, we fix each gene and then look at the KL divergence matrix of one cluster/subgroup $k$ relative to other cluster/subgroup $k^{'}$, which we call $KL^{g}_{K \times K}$. This matrix is symmetric and has all diagonal elements $0$ as the divergence of a cluster with respect to itself is $0$. Next we define the divergence measure for gene $g$ as 

$$ Div(g) = \underset{k}{max} \; \underset{l \neq k}{min} \; KL^{g} [k, l] $$

$$ K_{div}(g) = arg \;  \underset{k}{max} \; \underset{l \neq k}{min}  \; KL^{g} [k, l] $$


The higher the divergence measure, the more significant is the role of the gene in the clustering. We choose a small subset of around 50-100 genes with highest values of $Div(g)$ and put the gene in the $K_{div}(g)$ th cluster/subgroup. Then we perform gene annotations for the top genes in each subgroup using \textbf{mygene} R Bioconductor package \cite{Thompson2014}. We then try to see if the significant genes in a particular subgroup/cluster are associated with some specific biological functionality. This would indicate if the subgroups are actually biologically relevant or not. For instance, for GTEx tissue sample data, if the clusters are indeed driven by cell types, then the top genes for these clusters will probably be associated with proteins related to  functions for that particular cell type.


\medskip
















 









