
\begin{thebibliography}{9}

\bibitem{Sanders2014} 
S Anders, T P Pyl, W Huber.
\textit{HTSeq :  A Python framework to work with high-throughput sequencing data}. 
Bioinformatics, 2014, in print; online at doi:10.1093/bioinformatics/btu638

\bibitem{Li2009} 
Li H.*, Handsaker B.*, Wysoker A., Fennell T., Ruan J., Homer N., Marth G., Abecasis G., Durbin R. and 1000 Genome Project Data Processing Subgroup.
\textit{The Sequence alignment/map (SAM) format and SAMtools}. 
Bioinformatics, 2009, 25, 2078-9. [PMID: 19505943]

\bibitem{Liao2013}
Liao Y, Smyth GK and Shi W. 
\textit{The Subread aligner: fast, accurate and scalable read mapping by seed-and-vote}.
Nucleic Acids Research, 2013, 41, pp. e108.

\bibitem{Robinson2010}
Robinson MD, McCarthy DJ and Smyth GK. 
\textit{edgeR: a Bioconductor package for differential expression analysis of digital gene expression data}.
Bioinformatics, 2010, 26, pp. -1.

\bibitem{Ritchie2015}
Ritchie ME, Phipson B, Wu D, Hu Y, Law CW, Shi W and Smyth GK. 
\textit{limma powers differential expression analyses for RNA-sequencing and microarray studies}.
Nucleic Acids Research, 2015, 43(7), pp. e47.

\bibitem{Frazee2011}
Frazee AC, Langmead B, Leek JT. 
\textit{ReCount: a multi-experiment resource of analysis-ready RNA-seq gene count datasets}. 
BMC Bioinformatics,2011, 12:449

\bibitem{Zeisel2015}
Amit Zeisel, Ana B. Muñoz-Manchado, Simone Codeluppi, Peter Lönnerberg, Gioele La Manno, Anna Juréus, Sueli Marques, Hermany Munguba, Liqun He, Christer Betsholtz, Charlotte Rolny, Gonçalo Castelo-Branco, Jens Hjerling-Leffler, and Sten Linnarsson.
\textit{Cell types in the mouse cortex and hippocampus revealed by single-cell RNA-seq}.
Science 6 March 2015: 347 (6226), 1138-1142

\bibitem{GTEX2013}
The GTEx Consortium. 
\textit{The Genotype-Tissue Expression (GTEx) project}. 
Nature genetics. 2013;45(6):580-585. doi:10.1038/ng.2653.

\bibitem{Taddy2012}
Matt Taddy.
\textit{On Estimation and Selection for Topic Models}. 
AISTATS 2012, JMLR W\&CP 22.
(maptpx R package).

\bibitem{Pritchard2000}
Pritchard, Jonathan K., Matthew Stephens, and Peter Donnelly. 
\textit{Inference of population structure using multilocus genotype data}. 
Genetics 155.2 (2000): 945-959.

\bibitem{Raj2014}
Anil Raj, Matthew Stephens, and Jonathan K. Pritchard. 
\textit{fastSTRUCTURE: Variational Inference of Population Structure in Large SNP Data Sets.}
Genetics.2014 197:573-589.

\bibitem{Maaten2008}
L.J.P. van der Maaten and G.E. Hinton. 
\textit{Visualizing High-Dimensional Data Using t-SNE}. 
Journal of Machine Learning Research, 2008: 2579-2605.

\bibitem{Maaten2014}
L.J.P. van der Maaten. 
\textit{Accelerating t-SNE using Tree-Based Algorithms}. 
Journal of Machine Learning Research, 2014:3221-3245

\bibitem{Thompson2014}
Mark A, Thompson R and Wu C. 
\textit{mygene: Access MyGene.Info services}. 
2014. R package version 1.2.3.

\bibitem{Law2014}
Law CW, Chen Y, Shi W, Smyth GK. 
\textit{voom: precision weights unlock linear model analysis tools for RNA-seq read counts}. 
Genome Biology. 2014;15(2):R29. 

\bibitem{Jaitin2015}
Jaitin DA, Kenigsberg E et al.
\textit{Massively Parallel Single-Cell RNA-Seq for Marker-Free Decomposition of Tissues into Cell Types}.
Science, 2014: 343 (6172) 776-779 

\bibitem{Leek2010}
Jeffrey T. Leek, Robert B. Scharpf, Héctor C Bravo, David Simcha, Benjamin Langmead, W. Evan Johnson, Donald Geman, Keith Baggerly and Rafael A. Irizarry
\textit{Tackling the widespread and critical impact of batch effects in high-throughput data}.
Nature Reviews Genetics 11, 733-739

\bibitem{Hicks2015}
Stephanie C Hicks, Mingxiang Teng and Rafael A Irizarry
\textit{On the widespread and critical impact of systematic bias and batch effects in single-cell RNA-Seq data}.
BiorXiv, http://biorxiv.org/content/early/2015/09/04/025528






\end{thebibliography}

