\section{Results}

%An outline for results  (under consideration)
%
%\begin{itemize}
%
%\item Form two separate subsections, one for the GTEx Version 4 data and the other for the single cell Zeisel data.
%
%\item For GTEx data, give a figure comprising of 4 Structure plots for different $K$s, may be $2,5,10,15$. Fix the thinning parameter $p_{thin}$ to say $0.0001$.  Also record the log likelihoods (Bayes factors) for each of the 4 models, as reported by \textbf{maptpx}. 
%
%\item Have one figure showing the robustness of the clustering method on the thinning parameter $p_{thin}$. Fix $k=10$ and vary $p_{thin}$ to be $0.0001$, $0.001$ and $0.01$. 
%
%\item One t-SNE plot for GTEx samples (with and without admixture in the same plot). Should this be in results or in discussions? Also the t-SNE probably would require an electronic supplemental file as I would need the \textbf{qtlcharts} highlighting for those plots. 
%
%\item The GTEX brain samples Structure plot  for $K=4$ that shows the neuron cell types in brain cerebellum and cerebellar hemisphere. That is to show that the clusters are driven by cell types.
%
%\item Gene annotations for the GTEx significant genes (for brain) and also for the general set up (to decide on which $K$ to fix). Use Bayes Factor?
%
%\item The Structure plot for Zeisel single cell data. again Multiple $K=2,5,7,10$. 
%
%\item Gene annotations for the Zeisel single cell data. Need to choose the optimal $K$. Use Bayes Factor?
%
%\item t-SNE plot of the admixture proportions??..Is that required? Depends on how we present t-SNE. If this goes to discussion, we will avoid it here
%
%\end{itemize}

The admixture model was applied on the  V4 Genotype Tissue  Expression (GTEx) project  read counts data.  We first used the \textbf{eQtlBma} software due to Flutre \textit{et al} \cite{Flutre2013} to detect quantitative trait loci for gene expression (eQTLs) and then extracted all the cis gene -cis eQTL pairs from the data along with their posterior effect sizes for each tissue. We removed all gene-SNP pairs which had NA value reported in the posterior effect sizes for any of the tissues. Such NA values may result from ?? (small sample sizes, not all genotypes recorded?). At the end of the above filtering procedure, we were left with $16407$ cis-genes. We extracted out these genes from the reads matrix and applied our admixture model on the read counts matrix for $8555$ samples and these $16407$ cis-genes. We present the Structure plot for admixture model fit with $K=15$ in \textbf{Fig \ref{fig:fig1}}. The Structure plot reflects the homogeneity among the samples coming from the same tissue and also gives us an idea about which tissues have similar patterns of gene expression. For example, the different Brain tissues seem to cluster together, the same being true for the arteries (Artery-aorta, Artery-tibial and Artery-coronary). But interestingly, Muscle Skeletal and Heart tissues (Heart Left Ventricle and Heart Atrial Appendage) seem to be very close in their clustering patterns. The fact that in terms of gene expression patterns, Muscle Skeletal and Heart Left Ventricle are close is also evident from the fact that hierarchical clustering fails to separate out these two tissues \textbf{Fig \ref{fig:fig3}}. Besides the Structure plot, we feel another nice approach of visualizing the clustering patterns is using the t-SNE (Supplemental Fig \ref{fig:figS1}) \cite{Maaten2008} \cite{Maaten2014}. Using the expression matrix of the genes for the clusters (the $\theta$ matrix), we obtained the set of top driving genes for each cluster. In \textbf{Table \ref{tab:tab1}}, we present the gene names, the proteins they code and a short summary of their functions, obtained from the \textbf{mygene} package in R \cite{Thompson2014}. As can be seen from the table, \textit{PRM2}(protamine2), \textit{PRM1} (protamine1) and \textit{PHF7} (PHD finger protein 7) are the  top three genes that drive the cluster which separates out testis from the other tissues in the admixture cluster model in \textbf{Fig \ref{fig:fig1}}. Similarly, \textit{HBB} (hemoglobin, beta), \textit{HBA2} (hemoglobin, alpha 2) and \textit{HBA1} (hemoglobin, alpha 1) seem to be the top three genes that distinguish the whole blood and for a separate cluster from the rest. \\[3 pt]

A field of very active interest in recent times is to estimate the proportion of different cell types in different tissues. Marker based approaches are usually adopted to validate for different cell types and get a sense of the abundance of different cell types in the tissue samples \cite{Grun2015} \cite{Palmer2005}. The admixture model is a marker free method to obtain clusters driven by cell types. When applied on the full GTEX samples data, it is not evident because of the inter tissue variation is pretty strong. But when we apply the admixture model on just the Brain samples data, we see one cluster explaining around $80-85 \%$ admixture proportion in Brain Cerebellum and Cerebellar hemisphere (see \textbf{Fig \ref{fig:fig2}}) . This seems encouraging because it has been found using stereological approaches that rat cerebellum contains $> 80 \%$ neurons (Herculano-Houzel and Lent 2005) \cite{Houzel2005}. We subsequently performed gene annotations for a few top genes driving the different clusters (Supplementary Table 1) and observed that the pivotal genes that separated out this particular cluster in brain cerebellum and cerebellar hemisphere from the rest had synaptic activities. \\[3 pt]

Besides the biologically novel fact that admixture model seems generates clusters driven by cell types, evidence seems to suggest this method purely as a clustering technique seems to outperform the hierarchical clustering which is the most commonly used approach of clustering in RNA-seq and single cell seq literature. In \textbf{Fig \ref{fig:fig3}}, we consider every pair of tissues from the list of tissues in GTEX with number of samples $> 50$. Then we generated a set of $50$ samples randomly drawn from the pooled set of samples coming from these two tissues and then observed whether the hierarchical and the admixture were separating out samples coming from the two different tissues. For both the approaches, we used a color-coding scheme in case of complete separation of the two tissues in the pooled set of samples. We find that admixture model is more successful in separating out different tissues in general, compared to the hierarchical clustering technique. The admixture model is essentially a count based modeling approach and seems to handle low counts and zero counts much better than the hierarchical method which is a more general approach to clustering. Since the RNA-seq data and in particular single cell Seq data have lots of low counts and zero counts, the admixture model seems to be more suited for such data compared to hierarchical clustering method. \\[3 pt]

Currently there is a lot of interest in single cell sequencing as it is more informative about individual cell expression profiles compared to the RNA-seq on tissue samples. We were curious to see how stable the Admixture results are if the GTEx RNA-seq data is viewed at the scale of a single cell data. We achieve the latter by thinning the GTEx data under thinning parameter $p_{thin}=0.0001$ which is the order of scale obtained by dividing the total library size of the Jaitin \textit{et al} \cite{Jaitin2014} with respect to the library size of the GTEx V4 read counts data. We fitted the admixture model for $K=12$ on the thinned data and the Structure plot for the fitted model is presented in \textbf{Fig \ref{fig:fig4}}. It seems that most of the features observed in \textbf{Fig \ref{fig:fig1}} seem to be retained, for instance- the Brain samples clustering together, Whole blood and Testis forming separate clusters, Muscle skeletal and Heart tissue samples showing very similar patterns etc. However, thinning indeed shrinks the small differences across tissues and makes it more difficult to distinguish between tissues, as evident from the comparative study of hierarchical and admixture models, analogous to \textbf{Fig \ref{fig:fig3}}, for thinned data with thinning parameters $p_{thin}=0.001$ and $p_{thin}=0.0001$ in  \textbf{Fig \ref{fig:figS2}}. One can see that with thinning, the performance of admixture model in separating the tissues deteriorates but encouragingly, it seems that admixture does outperform the hierarchical clustering even under thinned data. \\[3pt]

Finally, we applied the admixture model on a couple of single cell datasets due to Jaitin \textit{et al} \cite{Jaitin2014} and Zeisel \textit{et al} \cite{Zeisel2015}.  Jaitin \textit{et al} sequenced around $4000$ single cells from mouse spleen, where the cells were a heterogeneous mix enriched for expression of CD11c marker.  The goal of their study was to separate out the B cells, NK cells, pDCs and monocytes. The sequencing was carried out in different amplification and sequencing batches. However the biological effect in their study was completely confounded with the amplification and sequencing batch effects. We present the Structure plot corresponding to the admixture model fit for $K=7$ for the Jaitin \textit{et al} data with the samples arranged by their amplification batch \textbf{Fig \ref{fig:fig5}} (top panel). Since the batch effects and  biological effects are confounded, it is difficult to interpret whether the clusters are driven by biology or by technical effects. Zeisel \textit{et al} analyzed the single cell data obtained from mouse cortex and hippocampus and obtained 47 molecularly distinct subclasses, comprising all known major cell types in the region. They also identified many marker genes informative about cell types, morphology and location. We fitted admixture model for $K=10$ on their data and we arranged the Structure samples as per their subclass assignment \textbf{Fig \ref{fig:fig5}} (bottom panel).The subclasses as depicted by them did seem to show pretty homogeneous patterns overall under Structure, but it was interesting that some of the samples in Oligo4 subclass seemed to show more heterogeneity - the proportion of red cluster seemed high for a few samples compared to others. Also the first few samples under Oligo6 seemed to show patterns similar to some of Oligo4 samples with lower red cluster proportion. These samples in Oligo6 were pretty different in pattern from the rest of Oligo6 samples which had no trace of red cluster. Since within each group, the samples are ordered in the same order as reported in the dataset, there is high likelihood, adjacent samples may be coming from same plate or may be sequenced in same lane etc, all of which can lead to similar patterns due to technical effects. The main highlight of \textbf{Fig \ref{fig:fig5}} is that one must be careful about interpreting Admixture results or any clustering results, as there is a possibility of batch effects driving the clusters instead of true biological effects. There has been a growing concern among biostatisticians today about how to deal with batch effects \cite{Leek2010} \cite{Hicks2015}. 








